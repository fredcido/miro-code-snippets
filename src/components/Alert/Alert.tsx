import {
  IconExclamationPointCircle,
  IconInformationMarkCircle,
  IconTickCircle,
} from "@mirohq/design-system";
import classNames from "classnames";
import React from "react";

export type Message = {
  variant: AlertVariant;
  content: string;
};

export type AlertVariant = "info" | "danger" | "success" | "warning" | "idle";

type Props = {
  icon?: React.ReactNode;
  children: React.ReactNode;
  variant: AlertVariant;
};

const variantProps = {
  info: {
    className:
      "text-blue-800 border-blue-300 bg-blue-50 dark:bg-gray-800 dark:text-blue-400 dark:border-blue-800",
    icon: <IconInformationMarkCircle />,
  },
  danger: {
    className:
      "text-red-800 border-red-300 bg-red-50 dark:bg-gray-800 dark:text-red-400 dark:border-red-800",
    icon: <IconExclamationPointCircle />,
  },
  warning: {
    className:
      "text-yellow-800 border-yellow-300 bg-yellow-50 dark:bg-gray-800 dark:text-yellow-400 dark:border-yellow-800",
    icon: <IconTickCircle />,
  },
  success: {
    className:
      "text-green-800 border-green-300 bg-green-50 dark:bg-gray-800 dark:text-green-400 dark:border-green-800",
    icon: <IconExclamationPointCircle />,
  },
  idle: {
    className:
      "text-gray-800 border-gray-300 bg-gray-50 dark:bg-gray-800 dark:text-gray-400 dark:border-gray-800",
    icon: <IconExclamationPointCircle />,
  },
} as const;

export function Alert({ children, variant, icon }: Props) {
  return (
    <div
      className={classNames(
        "flex items-center gap-3 rounded-lg border p-4",
        variantProps[variant].className
      )}
      role="alert"
    >
      {icon ?? variantProps[variant].icon}
      <div className="font-medium">{children}</div>
    </div>
  );
}
