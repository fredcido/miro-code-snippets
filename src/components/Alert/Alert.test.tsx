import { render } from "@testing-library/react";
import { Alert, type AlertVariant } from "./Alert";

describe("<Alert />", () => {
  test.each([
    { variant: "idle", content: "Hey Idle!" },
    { variant: "info", content: "From info!" },
    { variant: "danger", content: "Careful, danger!" },
    { variant: "success", content: "That worked" },
    { variant: "warning", content: "Watch out!" },
  ])("shoul render $variant with $content", ({ variant, content }) => {
    const { getByRole, container } = render(
      <Alert variant={variant as AlertVariant}>{content}</Alert>
    );
    expect(getByRole("alert")).toHaveTextContent(content);
    expect(container).toMatchSnapshot();
  });
});
