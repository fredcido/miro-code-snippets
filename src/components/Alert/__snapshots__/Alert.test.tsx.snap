// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Alert /> shoul render danger with Careful, danger! 1`] = `
<div>
  <div
    class="flex items-center gap-3 rounded-lg border p-4 text-red-800 border-red-300 bg-red-50 dark:bg-gray-800 dark:text-red-400 dark:border-red-800"
    role="alert"
  >
    <svg
      class="c-hdkwsG c-hdkwsG-dvzWZT-size-medium c-hdkwsG-OzWqL-weight-normal"
      fill="none"
      viewBox="0 0 24 24"
    >
      <path
        d="M12 8v4m9 0c0 4.9706-4.0294 9-9 9s-9-4.0294-9-9 4.0294-9 9-9 9 4.0294 9 9Zm-8.5 3.5a.5.5 0 1 1-1.0001-.0001.5.5 0 0 1 1.0001.0001Z"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-width="var(--svg-stroke-width)"
      />
      <path
        d="M13 15.5c0 .5523-.4477 1-1 1s-1-.4477-1-1 .4477-1 1-1 1 .4477 1 1Z"
        fill="currentColor"
      />
      <path
        d="M11.8097 12.3875v.0001"
        stroke="currentColor"
        stroke-width="var(--svg-stroke-width)"
      />
    </svg>
    <div
      class="font-medium"
    >
      Careful, danger!
    </div>
  </div>
</div>
`;

exports[`<Alert /> shoul render idle with Hey Idle! 1`] = `
<div>
  <div
    class="flex items-center gap-3 rounded-lg border p-4 text-gray-800 border-gray-300 bg-gray-50 dark:bg-gray-800 dark:text-gray-400 dark:border-gray-800"
    role="alert"
  >
    <svg
      class="c-hdkwsG c-hdkwsG-dvzWZT-size-medium c-hdkwsG-OzWqL-weight-normal"
      fill="none"
      viewBox="0 0 24 24"
    >
      <path
        d="M12 8v4m9 0c0 4.9706-4.0294 9-9 9s-9-4.0294-9-9 4.0294-9 9-9 9 4.0294 9 9Zm-8.5 3.5a.5.5 0 1 1-1.0001-.0001.5.5 0 0 1 1.0001.0001Z"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-width="var(--svg-stroke-width)"
      />
      <path
        d="M13 15.5c0 .5523-.4477 1-1 1s-1-.4477-1-1 .4477-1 1-1 1 .4477 1 1Z"
        fill="currentColor"
      />
      <path
        d="M11.8097 12.3875v.0001"
        stroke="currentColor"
        stroke-width="var(--svg-stroke-width)"
      />
    </svg>
    <div
      class="font-medium"
    >
      Hey Idle!
    </div>
  </div>
</div>
`;

exports[`<Alert /> shoul render info with From info! 1`] = `
<div>
  <div
    class="flex items-center gap-3 rounded-lg border p-4 text-blue-800 border-blue-300 bg-blue-50 dark:bg-gray-800 dark:text-blue-400 dark:border-blue-800"
    role="alert"
  >
    <svg
      class="c-hdkwsG c-hdkwsG-dvzWZT-size-medium c-hdkwsG-OzWqL-weight-normal"
      fill="none"
      viewBox="0 0 24 24"
    >
      <path
        d="M13 8.5c0 .5523-.4477 1-1 1s-1-.4477-1-1 .4477-1 1-1 1 .4477 1 1Z"
        fill="currentColor"
      />
      <path
        d="M12 12v4m9-4c0 4.9706-4.0294 9-9 9s-9-4.0294-9-9 4.0294-9 9-9 9 4.0294 9 9Zm-8.5-3.5a.5.5 0 1 1-1 0 .5.5 0 0 1 1 0Z"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-width="var(--svg-stroke-width)"
      />
      <path
        d="M11.8097 12.3875v.0001"
        stroke="currentColor"
        stroke-width="var(--svg-stroke-width)"
      />
    </svg>
    <div
      class="font-medium"
    >
      From info!
    </div>
  </div>
</div>
`;

exports[`<Alert /> shoul render success with That worked 1`] = `
<div>
  <div
    class="flex items-center gap-3 rounded-lg border p-4 text-green-800 border-green-300 bg-green-50 dark:bg-gray-800 dark:text-green-400 dark:border-green-800"
    role="alert"
  >
    <svg
      class="c-hdkwsG c-hdkwsG-dvzWZT-size-medium c-hdkwsG-OzWqL-weight-normal"
      fill="none"
      viewBox="0 0 24 24"
    >
      <path
        d="M12 8v4m9 0c0 4.9706-4.0294 9-9 9s-9-4.0294-9-9 4.0294-9 9-9 9 4.0294 9 9Zm-8.5 3.5a.5.5 0 1 1-1.0001-.0001.5.5 0 0 1 1.0001.0001Z"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-width="var(--svg-stroke-width)"
      />
      <path
        d="M13 15.5c0 .5523-.4477 1-1 1s-1-.4477-1-1 .4477-1 1-1 1 .4477 1 1Z"
        fill="currentColor"
      />
      <path
        d="M11.8097 12.3875v.0001"
        stroke="currentColor"
        stroke-width="var(--svg-stroke-width)"
      />
    </svg>
    <div
      class="font-medium"
    >
      That worked
    </div>
  </div>
</div>
`;

exports[`<Alert /> shoul render warning with Watch out! 1`] = `
<div>
  <div
    class="flex items-center gap-3 rounded-lg border p-4 text-yellow-800 border-yellow-300 bg-yellow-50 dark:bg-gray-800 dark:text-yellow-400 dark:border-yellow-800"
    role="alert"
  >
    <svg
      class="c-hdkwsG c-hdkwsG-dvzWZT-size-medium c-hdkwsG-OzWqL-weight-normal"
      fill="none"
      viewBox="0 0 24 24"
    >
      <g
        clip-path="url(#clip0_1908_1097)"
      >
        <path
          d="m8 12 3 3 5-6m5 3c0 4.9706-4.0294 9-9 9s-9-4.0294-9-9 4.0294-9 9-9 9 4.0294 9 9Z"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="var(--svg-stroke-width)"
        />
      </g>
      <defs>
        <clippath
          id="clip0_1908_1097"
        >
          <path
            d="M0 0h24v24h-24z"
            fill="none"
          />
        </clippath>
      </defs>
    </svg>
    <div
      class="font-medium"
    >
      Watch out!
    </div>
  </div>
</div>
`;
